ngraph/core/reference/src/runtime/reference/multiclass_nms.cpp

```diff
@@ -0,0 +1,514 @@
+                for (; idx < max_num_of_selected_indices; idx++)
+                {
+                    selected_indices_ptr[idx] = selected_index_filler;
+                    selected_scores_ptr[idx] = selected_score_filler;
+                }
+            }
+
+            void multiclass_nms_postprocessing(void* prois,
+                                               void* pscores,
+                                               void* pselected_num,
+                                               const ngraph::element::Type output_type,
+                                               const std::vector<float>& selected_outputs,
+                                               const std::vector<int64_t>& selected_indices,
+                                               const std::vector<int64_t>& valid_outputs,
+                                               const ngraph::element::Type selected_scores_type)
+            {
+                auto num_selected = std::accumulate(valid_outputs.begin(), valid_outputs.end(), 0);
+
+                /* data */
+                size_t selected_outputs_size = num_selected * 6;
+
+                switch (selected_scores_type)
+                {
+                case element::Type_t::bf16:
+                {
+                    bfloat16* scores_ptr = static_cast<bfloat16*>(prois);
+                    for (size_t i = 0; i < selected_outputs_size; ++i)
+                    {
+                        scores_ptr[i] = bfloat16(selected_outputs[i]);
+                    }
+                }
+                break;
+                case element::Type_t::f16:
+                {
+                    float16* scores_ptr = static_cast<float16*>(prois);
+                    for (size_t i = 0; i < selected_outputs_size; ++i)
+                    {
+                        scores_ptr[i] = float16(selected_outputs[i]);
+                    }
+                }
+                break;
+                case element::Type_t::f32:
+                {
+                    float* scores_ptr = static_cast<float*>(prois);
+                    memcpy(
+                        scores_ptr, selected_outputs.data(), selected_outputs_size * sizeof(float));
+                }
+                break;
+                default:;
+                }
+
+                if (pscores == nullptr)
+                {
+                    return;
+                }
+
+                size_t selected_indices_size = num_selected * 1;
+
+                if (output_type == ngraph::element::i64)
+                {
+                    int64_t* indices_ptr = static_cast<int64_t*>(pscores);
+                    memcpy(indices_ptr,
+                           selected_indices.data(),
+                           selected_indices_size * sizeof(int64_t));
+                }
+                else
+                {
+                    int32_t* indices_ptr = static_cast<int32_t*>(pscores);
+                    for (size_t i = 0; i < selected_indices_size; ++i)
```

Comment:
Maybe `for (int64_t i = 0; i < selected_outputs_size; ++i)`?

Reply:
Done